src/h5web/explorer/utils.ts

```diff
@@ -16,7 +16,7 @@ import { isMyGroup, isReachable } from '../providers/utils';
 function prepareEntity(
   link: HDF5HardLink | HDF5RootLink,
   parents: MyHDF5Group[]
-): Omit<MyHDF5Entity, 'kind'> {
+): Omit<MyHDF5Group | MyHDF5Dataset | MyHDF5Datatype, 'kind'> {
```

Comment:
Helps type check that the `id` is provided (a link doesn't have an `id`).

Reply:
In fact, nothing prevents us from attaching an `id` to links as well, making `id` required in `MyHDF5Entity`

Reply:
As discuss, let's think about this further. I'll merge as is for now.